File: my-folder/0080-remove-duplicates-from-sorted-array-ii/solution.c
int removeDuplicates(int* nums, int numsSize) {
    size_t j = 0;
    size_t seen = 1;
    for (size_t i = 1; i < numsSize; i++) {
        if (nums[i] == nums[i - 1]) {
            seen++;
        } else {
            seen = 1;
        }

        if (seen <= 2) {
            nums[++j] = nums[i];
        }
    }
    return j + 1;
}

File: my-folder/0080-remove-duplicates-from-sorted-array-ii/solution.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let mut seen = 0;

        nums.dedup_by(|a, b| {
            if a == b {
                seen += 1;
                if seen < 2 {
                    false
                } else {
                    true
                }
            } else {
                seen = 0;
                false
            }
        });

        nums.len() as i32
    }
}
